Sources/PaymentsCore/AnalyticsPayload.swift

```diff
@@ -0,0 +1,101 @@
+import UIKit
+
+struct AnalyticsPayload: Encodable {
+    
+    private let events: AnalyticsEvent
+    
+    init(eventName: String, sessionID: String) {
+        events = AnalyticsEvent(eventName: eventName, sessionID: sessionID)
+    }
+}
+
+private struct AnalyticsEvent: Encodable {
+    
+    let eventParams: AnalyticsEventParams
+    
+    init(eventName: String, sessionID: String) {
+        eventParams = AnalyticsEventParams(eventName: eventName, sessionID: sessionID)
+    }
+}
+
+private struct AnalyticsEventParams: Encodable {
```

Comment:
could we update this to either `AnalyticsEventParameters` or `AnalyticsEventDictionary` to make it a bit clearer than params?

Reply:
Another thing, Matt and I messed around with instead of having these multiple, nested struct types, which need to match the FPTI required structure of double nesting the `eventParams`, if we could use custom encoding to achieve the proper JSON format, while using just one `struct` type.

We got that to work and it looked like this:
```
    func encode(to encoder: Encoder) throws {
        var container = encoder.container(keyedBy: ContainerKeys.self)
        var events = container.nestedContainer(keyedBy: EventsKeys.self, forKey: .events)
        var eventParams = events.nestedContainer(keyedBy: EventParamsKeys.self, forKey: .eventParams)
        
        try eventParams.encode(eventName, forKey: .eventName)
    }
```

We decided that this was a little harder to read than just creating new struct types per level. But if others like this, we would only need one top-level struct type, `AnalyticsEventData` or whatever we call it.

Reply:
🤔  up to y'all - I don't feel like that's too difficult to parse but can see how it takes an extra bit of time to reason about if folks aren't familiar with nested containers.

Reply:
We opted to use the [custom encoder](https://github.com/paypal/iOS-SDK/pull/91/commits/f4c28d9a8557e38579e208c5e8a051f720b5b819)